translate: tools/index.md + tools/editor/* (#128)
OI-wiki/i18n: docs/tools/editor/geany.md

# Geany

Geany is a lightweight, easy-to-use text editor. It is friendly for beginners in Linux environment.

Like Dev-C++, it supports single file compilation.

However, it is compatible in Linux, Windows and macOS.

> The homepage of Geany is <https://geany.org/>

Reviewer: [Geany](https://geany.org) is a lightweight, easy-to-use text editor friendly for beginners in Linux environment. Like Dev-C++, it supports single file compilation but is compatible in Linux, Windows and macOS.

## Pros & Cons

### Pros

1. Lightweight;
2. Support for single file compilation;
3. No much need to configure;
4. Cross-platform.

### Cons

1. Not many users;

> 2. An access control issue was reported in macOS Catalina [^1].

Reviewer: permission?

> 3. No syntax highlighting by default when creating a new document. You may need to manually set filetype.

Reviewer: set->choose?

## Installation

### Windows/macOS

It is simple - just download installer from its official site and install it.

### Linux

#### Using Package Manager

Linux users can install Geany via package managers. For example, in debian-based distributions like Ubuntu or NOI Linux, run the following command and it will be automatically installed:

```bash
sudo apt install geany
```

#### Build from Source

1. Obtain sources from its repository or homesite
2. Run the following commands in terminal:

```bash
./configure
make
sudo make install
```

Note: you may need install a dependency `libgtk2.0-dev` if it reports `No package 'gtk+-2.0' found`.

> ## Skills

Reviewer: Tricks

### Switch filetype

Click `Document -> Set Filetype` to switch filetype.

For example, click `Document -> Set Filetype -> Programming Source -> C++ Source`, and you will see its syntax highlighting has been converted to C++.

### Set File Template

> Create a new folder `templates/files` under configuration directory, where files inside are template files. Restart Geany, and you will find `File -> New (With Template)` in menubar.

Reviewer: ```suggestion
Create a new folder `templates/files` under configuration directory, with template files inside. Restart Geany, and you will find `File -> New (With Template)` in menu bar.
```
